madmom/evaluation/chords.py

```diff
@@ -0,0 +1,969 @@
+    """
+    det_start = det_chords[0]['start']
+    ann_start = ann_chords[0]['start']
+    if det_start > ann_start:
+        filler = np.array((ann_start, det_start, chord('N')),
+                          dtype=CHORD_ANN_DTYPE)
+        det_chords = np.hstack([filler, det_chords])
+    elif det_start < ann_start:
+        det_chords = det_chords[det_chords['end'] > ann_start]
+        det_chords[0]['start'] = ann_start
+
+    det_end = det_chords[-1]['end']
+    ann_end = ann_chords[-1]['end']
+    if det_end < ann_end:
+        filler = np.array((det_end, ann_end, chord('N')),
+                          dtype=CHORD_ANN_DTYPE)
+        det_chords = np.hstack([det_chords, filler])
+    elif det_end > ann_end:
+        det_chords = det_chords[det_chords['start'] < ann_end]
+        det_chords[-1]['end'] = ann_chords[-1]['end']
+
+    return det_chords
+
+
+def segmentation(ann_starts, ann_ends, det_starts, det_ends):
+    """
+    Compute the normalized Hamming divergence between chord
+    segmentations as defined in [1]_ (Eqs. 8.37 and 8.38).
+
+    Parameters
+    ----------
+    ann_starts : list or numpy array
+        Start times of annotated chord segments.
+    ann_ends : list or numpy array
+        End times of annotated chord segments.
+    det_starts : list or numpy array
+        Start times of detected chord segments.
+    det_ends : list or numpy array
+        End times of detected chord segments.
+
+    Returns
+    -------
+    distance : float
+        Normalised Hamming divergence between annotated and
+        detected chord segments.
+
+    References
+    ----------
+    .. [1] Christopher Harte, "Towards Automatic Extraction of Harmony
+           Information from Music Signals." Dissertation,
+           Department for Electronic Engineering, Queen Mary University of
+           London, 2010.
+    """
+    est_ts = np.unique(np.hstack([det_starts, det_ends]))
+    seg = 0.
+    for start, end in zip(ann_starts, ann_ends):
+        dur = end - start
+        seg_ts = np.hstack([
+            start, est_ts[(est_ts > start) & (est_ts < end)], end])
+        seg += dur - np.diff(seg_ts).max()
+
+    return seg / (ann_ends[-1] - ann_starts[0])
+
+
+class ChordEvaluation(object):
+    """
+    Provide various chord evaluation scores.
+
+    Parameters
+    ----------
+    detections : str
+        File containing chords detections.
+    annotations : str
+        File containing chord annotations.
+    name : str
+        Name of the evaluation object (e.g., the name of the song).
+    """
+
+    METRIC_NAMES = [
+        ('root', 'Root'),
+        ('majmin', 'MajMin'),
+        ('majminbass', 'MajMinBass'),
+        ('sevenths', 'Sevenths'),
+        ('seventhsbass', 'SeventhsBass'),
+        ('segmentation', 'Segmentation'),
+        ('oversegmentation', 'OverSegmentation'),
+        ('undersegmentation', 'UnderSegmentation'),
+    ]
+
+    def __init__(self, detections, annotations, name, **kwargs):
```

Comment:
I am not sure if a `name` is really required. Most other evaluation classes don't do so. If you change it to `name=None`, please update the docstring accordingly.